### PR TITLE
Add integration tests and fix n+1 issue

### PR DESCRIPTION
**Pull request by m-talbi**
- Added integration tests for all `index` and `show` views.
- Fixed n+1 issue by using `includes` method

## Before
plus1
## After
Screenshot from 2023-03-06 19-02-12

### CODE REVIEW

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Kweeka1, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Meaningful commit messages
- Good README

## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/integration/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      visit user_posts_path('105')
+      expect(page).to have_content('pagination')
+    end
+
+    it 'User should see a post title' do
+      visit user_posts_path('105')
+      expect(page).to have_content('What does API mean?')
+    end
+  end
+end
```
Comment (DeliceLydia) [inline spec/integration/post_index_spec.rb]: - [ ] Please fix some of the tests that are failing
Screen Shot 2023-03-06 at 20 10 40

**Review by DeliceLydia — CHANGES_REQUESTED** [review]

**Comment by m-talbi** [comment]
- Please run these commands before running any test
```sh
bin/rails db:migrate:reset RAILS_ENV=test
```
```sh
bin/rails db:seed RAILS_ENV=test
```

**Review by mukhammadrizooff — APPROVED** [review]
Hi @Kweeka1, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Correct `GitFlow` ✔️ 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**